docs/lasagne.rst

```diff
@@ -49,14 +49,80 @@ Lasange to run:
   `lasagne/nolearn-based code
   <https://imatge.upc.edu/web/resources/end-end-convolutional-networks-saliency-prediction-software>`_.
 
+
+
+.. _layer-def:
+
+Neural Network Layer Definition
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+There are two supported methods of providing a Layer Definition to the :class:`.NeuralNet` constructor.
+The first involve generating the stack of :mod:`lasagne.layer` instances directly, while the second passes a dictionary
+and relies on the NeuralNet constructor to instantiate the layers.
+
+The sample network below is given as an example
+
+.. image:: LayerDefExample.svg
+
+Passing a Layer Instance Directly
+=================================
+The newer method is to simply set up a stack of layer instances, and pass the output layer to the constructor.
+This method is more versatile, and supports all types of :mod:`lasagne.layers`
+
+.. code-block:: python
+
+     from lasagne import layers
+
+     l_input = layers.InputLayer(shape=(None, 3, 32, 32), name="input")
+     l_conv = layers.Conv2DLayer(l_input, num_filters=16, filter_size=7, pad="same", stride=1, name="conv")
+     l_pool = layers.MaxPool2dLayer(l_conv, pool_size=2, stride=2, name="pool")
+     l_dense = layers.DenseLayer(l_pool, num_units=300, name="dense")
+
+    net = NeuralNet(layers=l_dense)
+
+Passing a dictionary of LayerTypes
+==================================
+The second method involves creating a diction where the key is the layer name, and the value is the layer type.
```

Comment:
I don't think this is accurate.  There's actually two alternatives to passing in a layer instance.  One which we don't need to mention is [still used in my tutorial](http://danielnouri.org/notes/2014/12/17/using-convolutional-neural-nets-to-detect-facial-keypoints-tutorial/#first-model-a-single-hidden-layer).  The other allows you to pass a list of tuples of the form:  `(layer_type, layer_kwargs)`.  An example of the latter, which we should document:

```python
nn = NeuralNet(
        layers=[
            (layers.InputLayer, {'shape': (None, 10)}),
            (layers.DenseLayer, {'num_units': 33}),
            ],
    )
```